src/scripts/main.js

```diff
@@ -1,3 +1,242 @@
+}
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setRandom() {
+  if (!hasEmptyTile()) {
+    return;
+  }
+
+  let findEmptyTile = false;
+
+  while (!findEmptyTile) {
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      board[r][c] = Math.random() > 0.9 ? 4 : 2;
+
+      const tile = document.getElementById(`${r}${c}`);
+
+      tile.innerText = `${board[r][c]}`;
+      tile.classList.add(`field-cell--${board[r][c]}`);
+      findEmptyTile = true;
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = '';
+  tile.classList.add('field-cell');
+
+  if (num > 0) {
+    tile.innerText = num;
+    tile.classList.add(`field-cell--${num}`);
+  }
+}
+
+const handler = (e) => {
+  switch (e.code) {
+    case 'ArrowLeft': {
+      slideLeftRight(true);
+      setTimeout(setRandom, 500);
+
+      break;
+    }
+
+    case 'ArrowRight': {
+      slideLeftRight(false);
+      setTimeout(setRandom, 500);
+
+      break;
+    }
+
+    case 'ArrowUp': {
+      slideUpDown(true);
+      setTimeout(setRandom, 500);
+
+      break;
+    }
+
+    case 'ArrowDown': {
+      slideUpDown(false);
+      setTimeout(setRandom, 500);
+
+      break;
+    }
+  }
+
+  loseGame();
+  winGame();
+};
+
+function tileWithoutZero(row) {
+  return row.filter(num => num !== 0);
+}
+
+function slide(row) {
+  let newRow = tileWithoutZero(row);
+
+  for (let i = 0; i < newRow.length; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow[i + 1] = 0;
+      score.textContent = `${+score.textContent + newRow[i]}`;
+    }
+  }
+
+  newRow = tileWithoutZero(newRow);
+
+  while (newRow.length < columns) {
+    newRow.push(0);
+  }
+
+  return newRow;
+}
+
+function slideLeftRight(left) {
+  for (let i = 0; i < rows; i++) {
+    let row = board[i];
+
+    if (left) {
+      row = slide(row);
+      board[i] = row;
+    } else {
+      row.reverse();
+      row = slide(row);
+      row.reverse();
+      board[i] = row;
+    }
+
+    for (let c = 0; c < columns; c++) {
+      const tile = document.getElementById(`${i}${c}`);
+      const num = board[i][c];
+
+      updateTile(tile, num);
+    }
+  }
+}
+
+function slideUpDown(up) {
+  for (let c = 0; c < columns; c++) {
+    let row = [
+      board[0][c],
+      board[1][c],
+      board[2][c],
+      board[3][c],
+    ];
+
+    if (up) {
+      row = slide(row);
+    } else {
+      row.reverse();
+      row = slide(row);
+      row.reverse();
+    }
+
+    for (let r = 0; r < columns; r++) {
+      board[r][c] = row[r];
+
+      const tile = document.getElementById(`${r}${c}`);
+      const number = board[r][c];
+
+      updateTile(tile, number);
+    }
+  }
+};
+
+function loseGame() {
```

Comment:
```suggestion
function checkIsGameLoss() {
```
you're doing a check here